### PR TITLE
feat(TDP-1489 TDQ-13265): add new action Fill empty cell from above

### PR DESCRIPTION
**Pull request by msjian**

### CODE REVIEW

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            this.value = value;
+        }
+    }
+
+    @Override
+    public void applyOnColumn(DataSetRow row, ActionContext context) {
+        // the first time applyOnColumn is called, save the current value in PreviousValueHolder
+        // and the Optional.ofNullable(...) allows you to NOT modify the first row (add an empty value)
+        // then second call of applyOnColumn, PreviousContextHolder has... previous value
+        final PreviousValueHolder holder = context.get("previous", values -> new PreviousValueHolder());
```
Comment (qiongli) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java]: I think this line is better to move 'compile(ActionContext context)'

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    @Override
+    public boolean acceptField(ColumnMetadata column) {
+        return true;
+    }
+
+    @Override
+    public Set<Behavior> getBehavior() {  return EnumSet.of(Behavior.VALUES_COLUMN, Behavior.FORBID_DISTRIBUTED); }
+
+    /** this class is used to store the previous value. */
+    private static class PreviousValueHolder {
```
Comment (hmdebenque) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java]: A static private class should be placed at the end of its containing class for easier readability

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    public void applyOnColumn(DataSetRow row, ActionContext context) {
+        // the first time applyOnColumn is called, save the current value in PreviousValueHolder
+        // and the Optional.ofNullable(...) allows you to NOT modify the first row (add an empty value)
+        // then second call of applyOnColumn, PreviousContextHolder has... previous value
+        final PreviousValueHolder holder = context.get("previous", values -> new PreviousValueHolder());
+        final String columnId = context.getColumnId();
+        final String value = row.get(columnId);
+
+        // Empty means null, empty string and any whitespace only strings
+        if (StringUtils.isBlank(value) || EMPTY_VALUE_LIST.contains(value)) {
```
Comment (hmdebenque) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java]: StringUtils.isBlank already rely on  Character.isWhitespace() which is a far more efficient algorithm and give the same result.

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            row.set(columnId, Optional.ofNullable(holder.getValue()).orElse(value));
```
Comment (hmdebenque) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java]: Why use Optionnals? Won't it be clearer to just add a `holder.getValue() != null` condition?
Or Objects.isNull() if you really want to make it nice.

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java (hunk 3 above)
Comment (msjian) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/FillEmptyFromAbove.java]: what we support white space is more then  Character.isWhitespace  support
you can see the description in https://jira.talendforge.org/browse/TDQ-13269

**Pull request merged**